src/base/preferences.cpp

```diff
@@ -449,6 +451,36 @@ void Preferences::setWebUiLocalAuthEnabled(bool enabled)
     setValue("Preferences/WebUI/LocalHostAuth", enabled);
 }
 
+bool Preferences::isWebUiAuthIPBypassEnabled() const
+{
+    return value("Preferences/WebUI/EnableAuthIPBypass", false).toBool();
+}
+
+void Preferences::setWebUiAuthIPBypassEnabled(bool enabled)
+{
+    setValue("Preferences/WebUI/EnableAuthIPBypass", enabled);
+}
+
+QList<QPair<QHostAddress, int> > Preferences::getWebUiAuthBypassSubnets() const
+{
+    QList<QPair<QHostAddress, int> > subnets;
+    QStringList rawSubnets = value("Preferences/WebUI/AuthBypassIP").toStringList();
+    foreach (const QString &rawSubnet, rawSubnets)
+        subnets.append(QHostAddress::parseSubnet(rawSubnet.toUtf8()));
+
+    return subnets;
+}
+
+void Preferences::setWebUiAuthBypassSubnets(const QList<QPair<QHostAddress, int> > &subnets)
+{
+    QStringList subnetsStringList;
+    QPair<QHostAddress, int> subnet;
```

Comment:
Why this var is outside the loop?

Reply:
I switched over to using the range-based for.